src/java/org/apache/cassandra/service/snapshot/SnapshotManifest.java

```diff
@@ -63,7 +63,7 @@ public SnapshotManifest(List<String> files, DurationSpec.IntSecondsBound ttl, In
     {
         this.files = files;
         this.createdAt = creationTime;
-        this.expiresAt = ttl == null ? null : createdAt.plusSeconds(ttl.toSeconds());
+        this.expiresAt = (ttl == null || ttl.toSeconds() == 0) ? null : createdAt.plusSeconds(ttl.toSeconds());
```

Comment:
leaving `ttl == null` here is fine and desirable as there is other code path (e.g. by taking snapshot) when passing null in e.g. jmx is just fine.